packages/application-generic/src/usecases/bulk-create-execution-details/bulk-create-execution-details.usecase.ts

```diff
@@ -19,9 +19,8 @@ export class BulkCreateExecutionDetails {
 
       entities.push(entity);
     });
-    // TODO: Which checks to do? If the notification and job belong to the environment and organization provided?
 
-    await this.executionDetailsRepository.insertMany(entities);
+    await this.executionDetailsRepository.insertMany(entities, true);
```

Comment:
[Continued](https://github.com/novuhq/novu/pull/3407/files#r1193123461) 
What I am not sure about here is the change of the behavior previously if we were inserting 100 documents and one failed. we still saved 99 documents. 
Now if we insert 100 documents and the first one failed all of them are not saved.
Did I got it right?

Could be related to storeJobs as well where if the first job is not saved(for example the email) all of the jobs won't be saved.

Reply:
Yes, it does mean that. It will save until there is an error in one. I'll think about it/another solution.

Reply:
I'd go with @scopsy suggestion. I think that is something we broadly miss in the DAL service. We tend to blindly trust bulk (and even single) inserts.

Reply:
agree, so we will do the `unordered` and we can check errors like this: https://www.mongodb.com/docs/manual/reference/method/db.collection.insertMany/#unordered-inserts